python/feathub/feature_views/derived_feature_view.py

```diff
@@ -107,6 +108,42 @@ def build(
             filter_expr=self.filter_expr,
         )
 
+    @staticmethod
+    def _validate(features: List[Feature], source: TableDescriptor) -> None:
+        # Check if the feature only depends on the features specified earlier or the
+        # features in the source table.
+        valid_variables = set([f.name for f in source.get_output_features()])
+        for feature in features:
+            transform = feature.transform
+            if isinstance(transform, JoinTransform):
+                variables = set()
```

Comment:
Is it possible to get the name of the table to be joined and consider its fields as well?

Reply:
The `variable` here refers to the input feature to compute the new feature. We then check that the `variable`  is a subset of `valid_variables`.  For JoinTransform, it doesn't has an input feature to compute the new feature. We only need to add the join feature to the `valid_variables`.